src/components/PersonLink/PersonLink.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Link, useLocation, useResolvedPath } from 'react-router-dom';
+import classNames from 'classnames';
+
+import { Person } from '../../types/Person';
+
+type Props = {
+  person: Person,
+};
+
+export const PersonLink: React.FC<Props> = React.memo(({ person }) => {
+  const {
+    name,
+    sex,
+    slug,
+  } = person;
+
+  const resolvedPath = useResolvedPath('../').pathname;
```

Comment:
The `useResolvedPath('../').pathname` might cause navigation issues. Consider using a more specific path or using `useRouteMatch` to get the current path.